core/src/main/java/cucumber/runtime/ThreadLocalRunnerSupplier.java

```diff
@@ -29,7 +33,7 @@ public ThreadLocalRunnerSupplier(
         this.backendSupplier = backendSupplier;
         this.runtimeOptions = runtimeOptions;
         this.glueSupplier = glueSupplier;
-        this.eventBus = eventBus;
+        this.eventBus = synchronize(eventBus);
```

Comment:
We know these busses will be called from different threads so synchronizing them in the supplier that creates them keeps these concerns together. Single threaded implementations can use a different supplier.

Reply:
Yeah sorry, my comment (deleted) was incorrect.